src/dotnet/CarbonAware/IntervalHelper.cs

```diff
@@ -0,0 +1,66 @@
+namespace CarbonAware;
+
+/// <summary>
+/// Helper for intervals within Carbon Aware.
+/// </summary>
+public static class IntervalHelper
+{
+
+    /// <summary>
+    /// Ensures the data is valid by using the min sampling window. Uses the expanded data to add an extra
+    /// data point before the startDate to ensure all expected data is included.
+    /// </summary>
+    /// <param name="data">Expanded data that includes the minimum sampling window</param>
+    /// <param name="startDate">Original start date provided by user</param>
+    /// <param name="endDate">Original end date provided by user</param>
+    /// <returns>Filtered emissions data.</returns>
+    /// <remarks> If the user provided interval is already >= minimum sampling window, or no expanded data, we just return the passed data. </remarks>
+    public static IEnumerable<EmissionsData> MinSamplingFiltering(IEnumerable<EmissionsData> expandedData, DateTimeOffset startDate, DateTimeOffset endDate, double minSamplingWindow)
+    {
+        TimeSpan ts = endDate - startDate;
+        if ((ts.TotalMinutes >= minSamplingWindow) || !expandedData.Any())
```

Comment:
Similar to my comment above, I think this logic might be somewhat flawed in that it will return any data that meets the size criteria without following through on the promise made by the summary: `Uses the expanded data to add an extra data point before the startDate to ensure all expected data is included`

Reply:
Updated version up that I think resolves this issue.